src/Abstractions/TaskOrchestrationContext.cs

```diff
@@ -364,18 +369,28 @@ public async Task<T> WaitForExternalEvent<T>(string eventName, TimeSpan timeout)
     /// Returns an instance of <see cref="ILogger"/> that is replay-safe, meaning that the logger only
     /// writes logs when the orchestrator is not replaying previous history.
     /// </summary>
-    /// <remarks>
-    /// This method wraps the provider <paramref name="logger"/> instance with a new <see cref="ILogger"/>
-    /// implementation that only writes log messages when <see cref="IsReplaying"/> is <c>false</c>.
-    /// The resulting logger can be used normally in orchestrator code without needing to worry about duplicate
-    /// log messages caused by orchestrator replays.
-    /// </remarks>
-    /// <param name="logger">The <see cref="ILogger"/> to be wrapped for use by the orchestration.</param>
-    /// <returns>An instance of <see cref="ILogger"/> that wraps the specified <paramref name="logger"/>.</returns>
-    public ILogger CreateReplaySafeLogger(ILogger logger)
-    {
-        return new ReplaySafeLogger(this, logger);
-    }
+    /// <param name="categoryName">The logger's category name.</param>
+    /// <returns>An instance of <see cref="ILogger"/> that is replay-safe.</returns>
+    public ILogger CreateReplaySafeLogger(string categoryName)
+        => new ReplaySafeLogger(this, this.LoggerFactory.CreateLogger(categoryName));
+
+    /// <summary>
+    /// Returns an instance of <see cref="ILogger"/> that is replay-safe, meaning that the logger only
+    /// writes logs when the orchestrator is not replaying previous history.
+    /// </summary>
+    /// <param name="type">The type to derive the category name from.</param>
+    /// <returns>An instance of <see cref="ILogger"/> that is replay-safe.</returns>
+    public virtual ILogger CreateReplaySafeLogger(Type type)
+        => new ReplaySafeLogger(this, this.LoggerFactory.CreateLogger(type));
+
+    /// <summary>
+    /// Returns an instance of <see cref="ILogger"/> that is replay-safe, meaning that the logger only
+    /// writes logs when the orchestrator is not replaying previous history.
+    /// </summary>
+    /// <typeparam name="T">The type to derive category name from.</typeparam>
+    /// <returns>An instance of <see cref="ILogger"/> that is replay-safe.</returns>
+    public virtual ILogger CreateReplaySafeLogger<T>()
```

Comment:
Consider using `inheritdoc` to remove the redundant `<summary>` and `<returns>` tags.